chef/cookbooks/crowbar/templates/default/crowbar-delete-Member-role-before-upgrade.sh.erb

```diff
@@ -0,0 +1,47 @@
+#!/bin/bash
+#
+# This script deletes the Member role as part of the upgrade from Cloud 8 to
+# Cloud 9. This is necessary because the "Member" role was renamed to "member"
+# in Cloud 9 and the attempt to create the "member" role will fail if "Member"
+# exists already (Keystone is case-insensitive).
+#
+# The script should be run only on the controller (first controller/cluster
+# founder in an HA cloud).
+
+LOGFILE=/var/log/crowbar/node-upgrade.log
+UPGRADEDIR=/var/lib/crowbar/upgrade
+STATEFILE=${UPGRADEDIR}/crowbar-delete-Member-role-before-upgrade-ok
+
+mkdir -p "`dirname "$LOGFILE"`"
+exec >>"$LOGFILE" 2>&1
+
+log()
+{
+    set +x
+    echo "[$(date --iso-8601=ns)] [$$] $@"
+    set -x
+}
+
+log "Executing $BASH_SOURCE"
+
+source /root/.openrc
+set -x
+
+mkdir -p $UPGRADEDIR
+
+if [[ -f $STATEFILE ]] ; then
+    log "Member role was already deleted"
+    exit 0
+fi
+
+if openstack role list | grep -q ' Member '; then
+  openstack role delete Member
```

Comment:
This has the side effect that all users with the Member role assignment on any project now lose that role assignment and won't be able to authenticate. Operators will have to recreate those role assignments manually and potentially update all of their policy files if they had been using 'Member' in their policies. Is this upgrade meant to be so disruptive?

Reply:
Oh...missed that problem. No the upgrade definitely isn't meant to be that disruptive. Then we're in a bit of a pickle. I see two potential solutions off the top of my head:

1. We could retain the old `Member` role and modify Crowbar to be case insensitive. Not too fond of that one...

2. We could extend `crowbar-delete-Member-role-before-upgrade.sh.erb` to save role assignments, create the `member` role right away and restore these role assignments. That's the approach I'd prefer.

Thoughts?

Reply:
For part 2, do you have idea how complicated could it be? Would you plan to automatically modify all possible policy files? Can't we miss something? Is it right to adapt what users set manually?

Reply:
I would go for 1 since we only support mysql backend now. Otherwise, I'd recommend doing SQL injection to rename 'Member' to 'member' so we can retain the role ID.

Reply:
How about this for option 1? https://github.com/crowbar/crowbar-openstack/pull/2153